[Yatoo2] iP

jtooya opened the pull request:
# Tayoo

> "For every minute spent in organizing, an hour is earned." - Benjamin Franklin

Yatoo frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/jtooya/ip/releases)
2. double-click it
3. add your tasks
4. let it manage your tasks for you :zany_face:

And it is **FREE**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jtooya committed:
Changed bot name
Added greeting and goodbye

jtooya committed:
Added echo

jtooya committed:
Added add, list

jtooya committed:
Refined awaitCommand() method
Added Task class along with markAsDone() and unmark() methods
Added mark and unmark commands

jtooya committed:
Added ToDo, Deadline and Event classes
Added todo, deadline and event commands

jtooya committed:
added text testing

jtooya committed:
Added error handling

jtooya committed:
Added delete command
Added NumberFormatException handling for
    mark, unmark and delete

jtooya committed:
Add javadocs comments for Deadline, Event, Task, Todo

jtooya committed:
Add capability to save tasks to .txt

jtooya committed:
Add function to mark/unmark tasks in tasklist.txt files

jtooya committed:
fix corrupted data file reading in parseTask()

jtooya committed:
Fix typo in readFromTaskList()

jtooya committed:
Merge branch 'branch-level-7' into Master

jtooya committed:
Rearrange imports to be alphabetical

jtooya committed:
Add LocalDateTime parsing for Deadline class

jtooya committed:
Add Test.java

jtooya committed:
Merge branch 'branch-Level-7' into Master

jtooya committed:
Fix typos

jtooya committed:
Add deleteTxt function

jtooya committed:
Add more OOP

jtooya committed:
Add time parser

jtooya committed:
Add time parsing

jtooya committed:
Merge tag 'A-MoreOOP'

jtooya committed:
Merge remote-tracking branch 'origin/master' into Branch-level-8

jtooya committed:
Add LocalDateTime parsing for Events

jtooya committed:
Merge branch 'Branch-Level-8'

jtooya committed:
Rearrange classes in packages

jtooya committed:
Merge branch 'add-gradle-support'

jtooya committed:
Add JUnit test for Tasklist and Parser

jtooya committed:
Edit build.gradle to include archiveVersion

jtooya committed:
Add JavaDoc header comments for all important classes

jtooya committed:
Merge branch 'A-JavaDoc'

jtooya committed:
Add find command

jtooya committed:
Merge branch 'branch-Level-9'

yuvrajaryan commented on src/main/java/tayoo/Parser.java:
```diff
@@ -0,0 +1,257 @@
+     * @throws ParserException if there is an error when parsing a command, either because the command is not supported,
+     * or the format of the command is wrong
+     */
+    public static Command parseCommand(String command) throws ParserException {
```
I think that you can consider naming the method like a verb so commandParser instead of parseCommand

yuvrajaryan commented on src/main/java/tayoo/Storage.java:
```diff
@@ -0,0 +1,199 @@
+        } catch (FileNotFoundException e) {
+            throw new UpdateTxtException("Cannot find tasklist.txt");
+        } catch (IOException e) {
+            throw new UpdateTxtException("An error occurred while reading the file");
```
I like that your names for exceptions are detailed

yuvrajaryan commented on src/main/java/tayoo/Storage.java:
```diff
@@ -0,0 +1,199 @@
+     * @param taskNumber the zero-based index of the task that is to be deleted within the tasklist.txt.
+     * @throws DeleteTxtException if tasklist.txt file could not be found, read or updated successfully.
+     */
+    public void deleteTxt(int taskNumber) throws DeleteTxtException {
```
I think that it would be better to spell out text instead of using txt

yuvrajaryan commented on src/main/java/tayoo/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,45 @@
+
+public class AddTaskCommand extends Command {
+
+    private final Task taskToAdd;
```
I like the descriptive name for the variable

yuvrajaryan reviewed (commented):
LGTM

jtooya committed:
Add checkStyle

jtooya committed:
Add gui

jtooya committed:
Fix event parser

jtooya committed:
Merge branch 'branch-Level-10'

jtooya committed:
Streamline deadline and event constructors

jtooya committed:
Enhance mark, unmark and delete command functionality to support marking/deleting multiple tasks at once

jtooya committed:
Add exit functionality to GUI

jtooya committed:
Merge branch 'master' into branch-Level-7

jtooya committed:
Test commit

jtooya committed:
Test commit

jtooya committed:
Merge branch 'branch-level-7'

jtooya committed:
Test commit

jtooya committed:
Merge branch 'branch-A-CodingStandard'

jtooya committed:
Test commit

jtooya committed:
Merge branch 'branch-level-10'

jtooya committed:
Add javadocs comments for all public methods

jtooya committed:
Add tests for all commands, parser and their dependencies

jtooya committed:
Add assertions to parser and tasklist

jtooya committed:
Improve code quality

jtooya committed:
Merge pull request #2 from jtooya/branch-A-Assertions

Add assertions to parser and tasklist

jtooya committed:
Merge branch 'master' into branch-A-CodeQuality

jtooya committed:
Merge pull request #3 from jtooya/branch-A-CodeQuality

Improve code quality

jtooya committed:
Create gradle.yml

jtooya committed:
Fix bugs in Storage class

jtooya committed:
Merge pull request #4 from jtooya/A-CI

Create gradle.yml

jtooya committed:
Polish GUI and chat bot appearance

The GUI is plain and lacks a sense of feedback for user input.

The resulting user experience is unsatisfying and uninspired as the
chat bot appears generic compared to other chat bots.

Let's
* Add colour chat bot's replies and vary the colour of the dialog box
according to the type of command given.
* Add colour to the text input field.
* Add a background image to separate this chat bot from others and make
the background image dark so that the dialog box stands out against the
background.
* Add a placeholder text in the text input field to more clearly
indicate to the user where the type their input.
* Add a title and name for the chat bot app.

jtooya committed:
Add JUnit tests for Tayoo#getResponse

There is no test for getResponse() in Tayoo to check that it sets the
field for isExit and commandType. If future commits alter the behaviour
of getResponse() or its dependent methods, or create new commandTypes
with incorrect return values for these variables, it could cause the
chat bot to malfunction.

Let's create a test to check that the getResponse() function works as expected.

That way, if any future changes to the code base alters either the
commands or getResponse, the JUnit tests will highlight it out to the
developer.

jtooya committed:
Add JUnit tests for Task, ToDo, Event and Deadline

There is currently no test to check that the implementation of toString
and toTxt work correctly for these classes. If the implementation of
either malfunctions due to future commits, it could cause a cascade of
errors. For example, the tasklist.txt would be interpreted as being corrupted because the format for parsing will be wrong.

Let's write some JUnit tests for these classes to ensure that the
conversion to String and txt representation is correct.

jtooya committed:
Add User Guide and Ui.png

jtooya committed:
Edit exit message for the chat bot

jtooya committed:
Fix bug in find command
